x/concentrated-liquidity/swaps_test.go

```diff
@@ -624,36 +643,52 @@ var (
 		//          5000
 		//  4545 -----|----- 5500
 		//  4545 -----|----- 5500
-		"two positions within one tick: usdc -> eth": {
-			tokenOut:                 sdk.NewCoin("usdc", sdk.NewInt(42000000)),
+		"two positions within one tick: eth (in) -> usdc (out) zfo": {
+			tokenOut:                 sdk.NewCoin("usdc", sdk.NewInt(66829187)),
 			tokenInDenom:             "eth",
-			priceLimit:               sdk.NewDec(5002),
+			priceLimit:               sdk.NewDec(4990),
 			swapFee:                  sdk.ZeroDec(),
 			secondPositionLowerPrice: DefaultLowerPrice,
 			secondPositionUpperPrice: DefaultUpperPrice,
-
-			expectedTokenOut:           sdk.NewCoin("usdc", sdk.NewInt(42000000)),
-			expectedTokenIn:            sdk.NewCoin("eth", sdk.NewInt(8398)),
-			expectedTick:               sdk.NewInt(310020),
-			expectedSqrtPrice:          sdk.MustNewDecFromStr("70.724513183069625078"), // https://www.wolframalpha.com/input?i=70.710678118654752440+%2B++++%2842000000++%2F+3035764687.503020836176699298%29
+			// from math import *
+			// from decimal import *
+			// liq = Decimal("1517882343.751510418088349649") * 2
+			// sqrt_cur = Decimal("5000").sqrt()
+			// token_out = Decimal("66829187")
+			// sqrt_next = sqrt_cur - token_out / liq
+			// token_in = token_in = liq * (sqrt_cur - sqrt_next) / (sqrt_cur * sqrt_next)
+			// print(sqrt_next)
+			// print(token_in)
+			expectedTokenOut:           sdk.NewCoin("usdc", sdk.NewInt(66829187)),
+			expectedTokenIn:            sdk.NewCoin("eth", sdk.NewInt(13370)),
+			expectedTick:               sdk.NewInt(309969),
+			expectedSqrtPrice:          sdk.MustNewDecFromStr("70.688664163727643650"),
 			expectedLowerTickFeeGrowth: DefaultFeeAccumCoins,
 			expectedUpperTickFeeGrowth: DefaultFeeAccumCoins,
 			// two positions with same liquidity entered
 			poolLiqAmount0: sdk.NewInt(1000000).MulRaw(2),
 			poolLiqAmount1: sdk.NewInt(5000000000).MulRaw(2),
 		},
-		"two positions within one tick: eth -> usdc": {
-			tokenOut:                 sdk.NewCoin("eth", sdk.NewInt(13370)),
+		"two positions within one tick: usdc (in) -> eth (out) ofz": {
+			tokenOut:                 sdk.NewCoin("eth", sdk.NewInt(8398)),
```

Comment:
Out of curiosity, what was the reasoning behind changing the `tokenOut` amount for these tests instead of keeping it the same and just changing the expected input value?

Reply:
I tried matching them with out given in tests. If you take a look at the corresponding test cases there, you will see that they almost match with some negligible variance.

This was helpful for benchmarking against the initial stages of the refactor